src/decoder/ifd.rs

```diff
@@ -15,18 +15,18 @@ use self::Value::{Ascii, List, Rational, Unsigned, Signed, SRational};
 #[allow(unused_qualifications)]
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub enum Value {
-    Signed(i32),
-    Unsigned(u32),
+    Signed(i64),
+    Unsigned(u64),
     List(Vec<Value>),
-    Rational(u32, u32),
-    SRational(i32, i32),
+    Rational(u64, u64),
+    SRational(i64, i64),
```

Comment:
Note that the converters `into_u32` etc. could then return an error if the representation is a BigTIFF value that happens to be out-of-range instead of completely replacing their declarations.

Reply:
I think using new variant for big tiff will add lot of code for not big gain :/
All the converter return an error if the representation is big tiff because they try to transform u32 to u64 with `try_from` so if the value is unreadable in u32 it will fail.

Reply:
> I think using new variant for big tiff will add lot of code for not big gain :/

It is the only remaining breaking change in this entire PR and we had _specifically_ made this enum non-exhaustive to not require it. From a maintainer perspective that gain is substantial on its own.

Further, the conversion will be adaptive. Offsets and values in big tiff are not _required_ to be larger than `u32` and those which are not can be handled just fine. This will effectively allows seemlessly handling files where the encoder carelessly always chooses big tiff regardless of whether it is necessary (I'm quite sure those exists and it's a good idea to follow Postel's law/Robustness principle). Also this is what the current implementation is doing anyways, with its use of `u32::try_from` below?

Reply:
It's done :wink: I am not a huge fan of the name of the enum var but didn't find better at the moment :smile: